config/initializers/sidekiq.rb

```diff
@@ -26,6 +27,7 @@
     end
 
     config.server_middleware do |chain|
+      chain.add Middleware::Sidekiq::OnlyRunWhileUp
```

Comment:
I wonder if there's a better name for this. "OnlyRunWhileUp" sounds obvious. How could it run while down? It's not until you get the further context of downtime detection.

Maybe `DowntimeDelayer` or `DowntimeDetector`?

Reply:
Yeah I have had some thoughts on a better name, but was just like get the MVP out the door.

Since it's not always delaying, I think `...Delayer` may be a bit misleading.

And since it can't detect the downtime, I think that's in the same ⛵ .

To talk it out, its main purpose is, if there are service downtimes to check, it checks whether the current time is inside one of the windows.

Hmmm, check...window...`DownTimeWindowChecker`?

Reply:
1️⃣  `ExternalServiceDowntimeChecker`
2️⃣  `ExternalServiceMaintenanceWindowChecker`
3️⃣  `DownTimeWindowChecker`
4️⃣  `ServiceDownTimeWindowChecker`

I like 3️⃣

Reply:
I am going to wait to make this name change in the morning. It's pretty late here and I've got everything buttoned up nicely. Thanks for all the help getting it stronger.

Maybe we can agree on a new name early tomorrow.

Reply:
It's been renamed to `Sidekiq::DowntimeWindowChecker`